lib/xboxrt/libc_extensions/direct.c

```diff
@@ -46,26 +46,28 @@ static int convert_error(DWORD winerror)
  */
 
 int _chdir(char* path) {
-    return 0;
+    assert(0);
+    return -1; // Unreachable
 }
 
 int _chdrive(int drive) {
-    return 0;
+    assert(0);
+    return -1; // Unreachable
 }
 
 char* _getcwd(char* buffer, size_t length) {
-    errno = -EINVAL;
-    return NULL;
+    assert(0);
+    return NULL; // Unreachable
 }
 
 char* _getwd(char* path_name) {
-    errno = -EINVAL;
-    return NULL;
+    assert(0);
+    return NULL; // Unreachable
```

Comment:
The macOS man page says:

> The function getwd() is a compatibility routine which calls getcwd() with its buf argument and a size of MAXPATHLEN (as defined in the include file <sys/param.h>).  Obviously, buf should be at least MAXPATHLEN bytes in length.

Similar wording is in POSIX https://pubs.opengroup.org/onlinepubs/009696699/functions/getwd.html

which should be trivial to implement.

I can't find anything about it in the MSDN docs though. Is it contained in the Windows SDK / MS implementations?

Reply:
Ah, yea, Wikipedia, and digitalmars state that getwd exists so I'm going on that
Should be trivial to implement though

Reply:
> I can't find anything about it in the MSDN docs though. Is it contained in the Windows SDK / MS implementations?

I don't see it in https://github.com/microsoft/win32metadata/blob/main/generation/WinSDK/RecompiledIdlHeaders/ucrt/direct.h either.

Reply:
Seems like getwd is a legacy function which was discontinued. The question is when did it get removed? As for digitalmars state it exists, I don't see getwd mentioned in the referenced link from the OP post. Heck, even did a search for whole site which also has no mention too.